tests/unittests/CoreGraphics/CGFontTests.mm

```diff
@@ -0,0 +1,117 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <TestFramework.h>
+#import <CoreGraphics/CoreGraphics.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+
+// CTFont functionality is generally a superset of CGFont functionality, and hits the same code path
+// Thus, CTFont unit tests can also be thought of as CGFont test coverage
+
+TEST(CGFont, GetFontBBox) {
+// Font names differ slightly between platforms
+#if TARGET_OS_WIN32
+    CFStringRef name = CFSTR("Arial Bold Italic");
+    CFStringRef name2 = CFSTR("Arial Italic");
+    CFStringRef name3 = CFSTR("Courier New Italic");
+#else
+    CFStringRef name = CFSTR("Arial-BoldItalicMT");
+    CFStringRef name2 = CFSTR("Arial-ItalicMT");
+    CFStringRef name3 = CFSTR("CourierNewPS-ItalicMT");
+#endif
+
+    // Windows font bounding boxes differ from ref plat's in 'interesting' but inconsistent ways
+    // Best hypothesis: Windows has different versions of the same fonts
+    CGFontRef font = CGFontCreateWithFontName(name);
+    CFAutorelease(font);
+    CGRect box = CGFontGetFontBBox(font);
+
+    // Eg: for Arial Bold Italic, origin is correct, but width and height are slightly off from ref plat (but in the right neighborhood)
+    EXPECT_EQ(-1146, box.origin.x);
+    EXPECT_EQ(-771, box.origin.y);
+#if TARGET_OS_WIN32
+    EXPECT_EQ(3993, box.size.width);
+    EXPECT_EQ(2855, box.size.height);
+#else
+    EXPECT_EQ(4196, box.size.width);
+    EXPECT_EQ(2820, box.size.height);
+#endif
+
+    font = CGFontCreateWithFontName(name2);
+    CFAutorelease(font);
+    box = CGFontGetFontBBox(font);
+
+    // For Arial Italic, only width is off from ref plat (but still in the right neighborhood)
+    EXPECT_EQ(-1059, box.origin.x);
+    EXPECT_EQ(-665, box.origin.y);
+    EXPECT_EQ(2708, box.size.height);
+#if TARGET_OS_WIN32
+    EXPECT_EQ(3842, box.size.width);
+#else
+    EXPECT_EQ(3884, box.size.width);
+#endif
+
+    font = CGFontCreateWithFontName(name3);
+    CFAutorelease(font);
+    box = CGFontGetFontBBox(font);
+
+// For Courier New Italic, both horizontal attributes are off by exactly 115, and both vertical attributes are off by exactly 9
+#if TARGET_OS_WIN32
+    EXPECT_EQ(-138, box.origin.x);
```

Comment:
Didn't actually check but there's a 0% chance in my mind that it could look different at all. This API returns the total accumulated bounding box of every glyph in the font. As you can see below, we get correct bounding boxes for individual glyphs, so it's most likely that one version of the font has some extra glyphs.